step-api/src/main/java/org/jenkinsci/plugins/workflow/structs/DescribableHelper.java

```diff
@@ -331,44 +341,40 @@ static ParameterType of(Type type) {
                     }
                 }
                 if (acceptsList(type)) {
-                    return new ArrayType(of(((ParameterizedType) type).getActualTypeArguments()[0]));
+                    return new ArrayType(type, of(((ParameterizedType) type).getActualTypeArguments()[0]));
                 }
                 throw new UnsupportedOperationException("do not know how to categorize attributes of type " + type);
             } catch (Exception x) {
-                return new ErrorType(x);
+                return new ErrorType(x, type);
             }
         }
     }
 
     public static final class AtomicType extends ParameterType {
-        private final Class<?> clazz;
         AtomicType(Class<?> clazz) {
-            this.clazz = clazz;
+            super(clazz);
         }
-        /**
-         * A Java type: either {@link String}, or a {@link Class#isPrimitive} wrapper.
-         */
+
```

Comment:
Of course now you have come full circle and deleted my nice Javadoc. :-) No matter.